pkg/query-service/app/logs/v3/enrich_query_test.go

```diff
@@ -393,6 +394,71 @@ var testEnrichParamsData = []struct {
 			},
 		},
 	},
+	{
+		Name: "Enrich if an attribute/resource attribute is materialized/dematerialized",
+		Params: v3.QueryRangeParamsV3{
+			CompositeQuery: &v3.CompositeQuery{
+				BuilderQueries: map[string]*v3.BuilderQuery{
+					"test": {
+						QueryName:  "test",
+						Expression: "test",
+						DataSource: v3.DataSourceLogs,
+						AggregateAttribute: v3.AttributeKey{
+							Key:      "mat_resource",
+							Type:     v3.AttributeKeyTypeResource,
+							DataType: v3.AttributeKeyDataTypeInt64,
+							IsColumn: true,
+						},
+						Filters: &v3.FilterSet{Operator: "AND", Items: []v3.FilterItem{
+							{Key: v3.AttributeKey{Key: "mat_attr", Type: v3.AttributeKeyTypeTag, IsColumn: true}, Value: "test", Operator: "="},
+							{Key: v3.AttributeKey{Key: "normal_attr", DataType: v3.AttributeKeyDataTypeString, IsColumn: false}, Value: "test1", Operator: "="},
+						}},
+					},
+				},
+			},
+		},
+		Fields: map[string]v3.AttributeKey{
+			"mat_resource": {
+				Key:      "mat_resource",
+				Type:     v3.AttributeKeyTypeResource,
+				DataType: v3.AttributeKeyDataTypeInt64,
+				IsColumn: false,
+			},
+			"mat_attr": {
```

Comment:
fields being map structure means only one item with a key can exist. what happens if there are conflicting types, the user selection should be preferred?

Reply:
This condition takes care tat if the type doesn't match it won't update https://github.com/SigNoz/signoz/blob/0711c8855ea5c795202ea68fde9a75cf05119b1b/pkg/query-service/app/logs/v3/enrich_query.go#L150 .

but let me fix this in a better way

Reply:
have made some changes please, have a look I have also updated the description.